### PR TITLE
Extract Onboarding into it's own controller

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Refactor

## Description
Create OnboardingController with proper authenticated & spec.

## Related Tickets & Documents
Resolves https://github.com/thepracticaldev/dev.to/issues/6602

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a

## Added tests?
- [x] yes

## Added to documentation?
- [x] no documentation needed

## [optional] Are there any post deployment tasks we need to perform?
n/a

### CODE REVIEW

File: app/controllers/onboardings_controller.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class OnboardingsController < ApplicationController
```
Comment (rhymes) [inline app/controllers/onboardings_controller.rb]: should this be a singular name, `OnboardingController` ? It's declared as a `resource`, not `resources` in the routes file

File: spec/requests/onboardings_spec.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+require "rails_helper"
```
Comment (rhymes) [inline spec/requests/onboardings_spec.rb]: same here, maybe bot the file and the description should be singular

File: spec/requests/onboardings_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+RSpec.describe "Onboardings", type: :request do
+  let(:user) { create(:user, saw_onboarding: false) }
+
+  describe "GET /onboarding" do
+    it "redirects user if unauthenticated" do
+      get onboarding_url
+      expect(response).to redirect_to("/enter")
+    end
+
+    it "return 200 when authentidated" do
```
Comment (rhymes) [inline spec/requests/onboardings_spec.rb]: small typo here

**Comment by maestromac** [comment]
Oh I was going to update the typo but thanks for the merge!

**Pull request merged**